### PR TITLE
Fix: immediate update for path

### PR DESCRIPTION
**Pull request by rma4ok**
* Running immediate updates with path without debouncing

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Branch being merged does not follow [Git Flow](http://nvie.com/posts/a-successful-git-branching-model/)
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+13842 -13)  
Please keep it below 500 net changes</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d513d3720ec595a5fcef42f6c6d47da22d573ae0
</p>

**Review by daedalus28 — CHANGES_REQUESTED** [review]
This needs test cases

File: src/index.js [hunk 1]
```diff
@@ -146,12 +146,18 @@ export let ContextTree = _.curry(
...
+      F.debounceAsync(debounce, runUpdate)
+    )
+
     let triggerUpdate = path =>
-      TreeInstance.disableAutoUpdate
-        ? triggerImmediateUpdate(path)
-        : triggerDelayedUpdate(path)
+      (TreeInstance.disableAutoUpdate
+        ? triggerImmediatePathUpdate
+        : triggerDelayedPathUpdate)(String(path))(path)
```
Comment (daedalus28) [inline src/index.js:160]: What is `String(path)` for? if you're trying to do work around mobx observables, the way to do this is using `snapshot` from config (which is `toJS` in a mobx context)

File: src/index.js [hunk 2]
```diff
@@ -146,12 +146,18 @@ export let ContextTree = _.curry(
         onError(error) // Raise the onError event
       }
     }
-    let triggerImmediateUpdate = F.debounceAsync(1, runUpdate)
-    let triggerDelayedUpdate = F.debounceAsync(debounce, runUpdate)
+
+    let triggerImmediatePathUpdate = _.memoize(() =>
```
Comment (daedalus28) [inline src/index.js:150]: why do we need to memoize these? This only runs once when you instantiate a client instance anyway

**Review by daedalus28 — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (rma4ok) [reply src/index.js:150]: we need to isolate debouncing for different paths
you if you refresh root and then unpause a facet, second update will bounce out the root refresh
so using memo for different debouncers

Comment (daedalus28) [reply src/index.js:150]: Let's just add a comment here explaining this

File: src/index.js (hunk 1 above)
Comment (rma4ok) [reply src/index.js]: replaced with `encode`

**Review by daedalus28 — APPROVED** [review]

**Pull request merged**